homeassistant/components/mqtt/__init__.py

```diff
@@ -418,35 +435,50 @@ def async_publish_service(call):
     return True
 
 
-Subscription = namedtuple('Subscription',
-                          ['topic', 'callback', 'qos', 'encoding'])
-Subscription.__new__.__defaults__ = (0, 'utf-8')
+@attr.s(slots=True, frozen=True)
+class Subscription(object):
+    """Class to hold data about an active subscription."""
```

Comment:
`Subscription` and `Message` objects now use the new attrs library (home-assistant/architecture#4).